Update publish-and-synchronization.md (#2659)
spryker/spryker-docs: docs/dg/dev/backend-development/data-manipulation/data-publishing/publish-and-synchronization.md

	}
}
```

### Data Architecture 

> P&S plays a major role in data denormalization and distribution to Spryker storefronts and API. Denormalization procedure aims for preparing data in the form it will be consumed by data clients. Distribution procedure aims to distribute the data closer to the end users, so that they feel like accessing a local storage.

Reviewer: ```suggestion
P&S plays a major role in data denormalization and distribution to Spryker storefronts and API.
- Denormalization procedure aims for transforming data to the format that will be consumed by data clients.
- Distribution procedure aims to distribute data, so the users feel like accessing a local storage.
```

Author: I want to keep it as a paragraph. Not everything should be a bulletpoint)

{% info_block infoBox "Project Example"%}

> Some of Spryker partners applied P&S features to get data distributed over picking devices (barcode scanners) in a warehouse. Which accessed the Spryker catalog as their local one, even with low quality or missing network access.

Reviewer: ```suggestion
Some of Spryker partners applied P&S features to distribute data over picking devices (barcode scanners) in a warehouse. The picking devices accessed the Spryker catalog as their local one, enabling them to properly behave with low quality or missing network access.
```

> In order cases, P&S enabled customer projects to keep their main backend systems with customer data in one region (eg Germany), while providing local catalogs over the world. This enabled them on one hand to keep customer data under data privacy constraints, and on other hand their buyers in Brazil did not feel an oversea long response times, browsing their catalogs as "local" with blazing fast response times.

Reviewer: ```suggestion
In other cases, P&S enabled customer projects to keep their main backend systems with customer data in one region (eg Germany), while distributing local catalogs over the world. This enabled them on one hand to keep customer data under data privacy constraints, and on other hand their buyers in Brazil can browse their catalogs (as "local") with blazing fast response times.
```

> Indeed, P&S inspires intelegent solutions and smart architecture designs!

Reviewer: ```suggestion
P&S inspires intelligent solutions and smart architecture designs!
```

{% endinfo_block %}

> When desining a solution using P&S we need to consider the following concerns in our applications

Reviewer: ```suggestion
When designing a solution using P&S we need to consider the following concerns in our applications
```

- eventual consistency for data available in storefronts

> - horizontal scalability of Publish process (native) and Sync process (requires development)

Reviewer: ```suggestion
- horizontal scalability of publish process (native) and sync process (requires development)
```

- data object limitations

#### Data Object Limitations

> In order to build a healthy commerce system, we need to make sure that P&S process is healthy at all times. And first we start with a healthy NFRs for P&S.

Reviewer: ```suggestion
In order to build a healthy commerce system, we need to make sure that P&S process is healthy at all times. And first we start with healthy NFRs for P&S.
```

> - storage sync message size should not be over 256Kb - this prevents us from problems in data processing, but even more important in data comsumption, when an API consumer might experience failure when reviceing an aggregated object of a high size.

Reviewer: ```suggestion
- storage sync message size should not be over 256Kb - this prevents some problems in the store during data processing, and especially during data consumption (when an API consumer might experience failure when receiving an aggregated object of a huge size).
```

Author: ```suggestion
- storage sync message size should not be over 256KB - this prevents us from some problems in data processing, but even more important in data consumption, when an API consumer might experience failure when receiving an aggregated object of a large size.
```

> - do not exceed the request limitations for the storage (eg. Redis) and search (eg. OpenSearch) systems, while sending data in sync process

Reviewer: ```suggestion
- do not exceed the batch limitations for the storage (eg. Redis) and search (eg. OpenSearch) systems, while sending data in sync process
```

Reviewer: This comment can be ignored, as you prefer. I've approved the PR in general.

> - do not exceed the request limitations for the storage (eg. Redis) and search (eg. OpenSearch) systems, while sending data in Sync process

Reviewer: ```suggestion
- do not exceed the request limitations for the storage (eg. Redis) and search (eg. OpenSearch) systems, while sending data in sync process
```

> - do not exceed the request limitations for the storage (eg. Redis) and search (eg. OpenSearch) systems, while sending data in Sync process

Reviewer: How is this different than the previous NFR?

Author: What are the prevous NFR?

Reviewer: This is the current NFR, I have commented on: `do not exceed the request limitations for the storage (eg. Redis) and search (eg. OpenSearch) systems, while sending data in Sync process` 

And this is the previous NFR: `storage sync message size should not be over 256Kb - this prevents us from problems in data processing, but even more important in data comsumption, when an API consumer might experience failure when reviceing an aggregated object of a high size.`

I don't see any difference in the meaning, but 2 versions of the same semantical meaning.

{% info_block infoBox "Are these really limitations?"%}

As every non-functional requirement (NFR) each of these limitations might be changed and adjusted to the project needs.
However that might require an additional implementation or refactorring of Spryker OOTB functionallities.

> For examples if our business requirements include sending objects over 10Mb via API, which is not typical for ecommerce projects, that is still possible with Spryker. However it might require a review of the business logic attached to the API endpoint, and be adjusted to the new requirements when neccesary. In this case the default requirment of 256Kb for a sync message size is not applicable anymore.

Reviewer: ```suggestion
For examples if your business requirements include sending objects above 10MB via API, which is not typical for e-commerce projects, that is still possible with Spryker. However it might require a review of the business logic attached to the API endpoint, and be adjusted to the new requirements. In this case the default requirement of 256KB for a sync message size is not applicable anymore.
```
